pkg/mermaid/sequencediagram/sequencediagram_test.go

```diff
@@ -0,0 +1,85 @@
+package sequencediagram
+
+import (
+	"testing"
+
+	"github.com/alecthomas/assert"
+	"github.com/anz-bank/sysl/pkg/parse"
+	"github.com/anz-bank/sysl/pkg/syslutil"
+	"github.com/spf13/afero"
+)
+
+func TestBadInputsToGenerateMermaidSequenceDiagram(t *testing.T) {
+	t.Parallel()
+	appName := "wrongName"
+	epName := "wrongEp"
+	m, err := parse.NewParser().Parse("demo/simple/sysl-sd.sysl",
+		syslutil.NewChrootFs(afero.NewOsFs(), projectDir))
+	if err != nil {
+		t.Error(err)
+	}
+	r, err := GenerateSequenceDiagram(m, appName, epName)
+	assert.NotNil(t, m)
+	assert.Empty(t, r)
+	assert.Error(t, err)
+}
+
+func TestGenerateMermaidSequenceDiagram(t *testing.T) {
+	t.Parallel()
+	appName := "WebFrontend"
+	epName := "RequestProfile"
+	m, err := parse.NewParser().Parse("demo/simple/sysl-sd.sysl",
+		syslutil.NewChrootFs(afero.NewOsFs(), projectDir))
+	if err != nil {
+		t.Error(err)
+	}
+	r, err := GenerateSequenceDiagram(m, appName, epName)
+	assert.NotNil(t, m)
+	assert.NotNil(t, r)
+	assert.Nil(t, err)
+}
+
+func TestGenerateMermaidSequenceDiagram2(t *testing.T) {
+	t.Parallel()
+	appName := "WebFrontend"
+	epName := "RequestProfile"
+	m, err := parse.NewParser().Parse("demo/simple/sysl-sd2.sysl",
+		syslutil.NewChrootFs(afero.NewOsFs(), projectDir))
+	if err != nil {
+		t.Error(err)
+	}
+	r, err := GenerateSequenceDiagram(m, appName, epName)
+	assert.NotNil(t, m)
+	assert.NotNil(t, r)
+	assert.Nil(t, err)
+}
+
+func TestGenerateMermaidSequenceDiagram3(t *testing.T) {
+	t.Parallel()
+	appName := "MobileApp"
+	epName := "Login"
+	m, err := parse.NewParser().Parse("demo/simple/sysl-app-hyperlink.sysl",
+		syslutil.NewChrootFs(afero.NewOsFs(), projectDir))
+	if err != nil {
+		t.Error(err)
+	}
+	r, err := GenerateSequenceDiagram(m, appName, epName)
+	assert.NotNil(t, m)
+	assert.NotNil(t, r)
+	assert.Nil(t, err)
+}
+
+func TestGenerateMermaidSequenceDiagramWithIfElseLoopActionAndGroupStatements(t *testing.T) {
+	t.Parallel()
+	appName := "DragonEater"
+	epName := "EatDragon"
+	m, err := parse.NewParser().Parse("demo/simple/best-ever-sysl-example.sysl",
+		syslutil.NewChrootFs(afero.NewOsFs(), projectDir))
+	if err != nil {
+		t.Error(err)
+	}
+	r, err := GenerateSequenceDiagram(m, appName, epName)
+	assert.NotNil(t, m)
+	assert.NotNil(t, r)
+	assert.Nil(t, err)
```

Comment:
These testcases still don't do much

Reply:
Yes, it did when I first implemented it, I have removed a good chunk of it. Let me see if I can add anything to it but for now, it helps me make sure that I have not messed up something really bad.